ludwig/data/preprocessing.py

```diff
@@ -1647,32 +1648,45 @@ def preprocess_for_training(
         with backend.storage.cache.use_credentials() if cached else contextlib.nullcontext():
             logger.debug("create training dataset")
             training_dataset = backend.dataset_manager.create(training_set, config, training_set_metadata)
-            if not len(training_set):
+            training_set_size = len(training_dataset)
+            if training_set_size == 0:
                 raise ValueError("Training data is empty following preprocessing.")
+            elif training_set_size < MIN_DATASET_SPLIT_ROWS:
+                raise ValueError(f"Training dataset has only {training_set_size} rows following preprocessing.")
 
             validation_dataset = None
             if validation_set is not None:
                 logger.debug("create validation dataset")
                 validation_dataset = backend.dataset_manager.create(validation_set, config, training_set_metadata)
-                if not len(validation_dataset):
-                    # Validation dataset is empty.
+                validation_set_size = len(validation_dataset)
+                if validation_set_size == 0:
                     logger.warning(
-                        "Encountered empty validation dataset. If this is unintentional, please check the "
-                        "preprocessing configuration."
+                        "Validation set empty. If this is unintentional, please check the preprocessing configuration."
+                    )
+                    validation_dataset = None
+                elif validation_set_size < MIN_DATASET_SPLIT_ROWS:
+                    logger.warning(
+                        f"Validation set ({validation_set_size}) too small to compute metrics. Evaluation of validation"
+                        " set will be skipped"
```

Comment:
Would it help to tell the user the number of rows it has and the number of rows it should have to not skip validation?